common/include/window.h

```diff
@@ -50,6 +50,79 @@ static inline void set_embedded_window_pointer(embed_window_pointer_t *wp, windo
 
 static inline void set_embedded_window_pointer(ignore_window_pointer_t *, window *) {}
 
+struct window
```

Comment:
You could use *class-key* `class` to introduce the type and set the default member access to `private`, which would remove the need to specify `private:` two lines down.  You can leave this as-is if you like.  It runs the same either way.